packages/lesswrong/server/vulcan-lib/apollo-ssr/pageCache.ts

```diff
@@ -81,13 +81,26 @@ type InProgressRender = {
 
 const inProgressRenders: Record<string,Array<InProgressRender>> = {};
 
+function filterLoggedOutActiveAbTestGroups(abTestGroups: CompleteTestGroupAllocation): RelevantTestGroupAllocation {
+  const abTests = getABTestsMetadata();
+  let result: RelevantTestGroupAllocation = {};
+  for (const name of Object.keys(abTestGroups)) {
+    const abTest = abTests[name];
+    if (abTest.active && abTest.affectsLoggedOut)
+      result[name] = abTestGroups[name];
+    
+  }
+  return result;
+}
+
 // Serve a page from cache, or render it if necessary. Takes a set of A/B test
 // groups for this request, which covers *all* A/B tests (including ones that
 // may not be relevant to the request).
 export const cachedPageRender = async (req: Request, abTestGroups: CompleteTestGroupAllocation, userAgent: string|undefined, renderFn: (req:Request)=>Promise<RenderResult>) => {
   const path = getPathFromReq(req);
   const cacheKey = cacheKeyFromReq(req);
-  const cached = await cacheLookup(cacheKey, abTestGroups);
+  const cacheAffectingAbTestGroups = filterLoggedOutActiveAbTestGroups(abTestGroups);
+  const cached = await cacheLookup(cacheKey, cacheAffectingAbTestGroups);
```

Comment:
I think there's some type mismatch happening here - `filterLoggedOutActiveAbTestGroups()` returns a `RelevantTestGroupAllocation` but `cacheLookup()` expects `CompleteTestGroupAllocation`. I think it doesn't cause a bug in this case but I'm not sure, and probably these two types should be more distinct to avoid this kind of thing.